### ISSUE TITLE
Could not pass event AsyncPlayerChatEvent to PwnFilter

### ISSUE THREAD

**Issue by mibby** [Pwn9/PwnFilter#19]
PwnFilter dev 101
PaperSpigot dev 691

```
[16:26:31] [Async Chat Thread - #4/ERROR]: Could not pass event AsyncPlayerChatEvent to PwnFilter v3.9.1-SNAPSHOT-b101
com.google.common.util.concurrent.ExecutionError: java.lang.InstantiationError
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2199) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache.get(LocalCache.java:3934) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3938) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4821) ~[patched_1.9.2.jar:git-Paper-691]
	at com.pwn9.filter.bukkit.BukkitAPI.getAuthorById(BukkitAPI.java:75) ~[?:?]
	at com.pwn9.filter.bukkit.BukkitAPI.getAuthorById(BukkitAPI.java:46) ~[?:?]
	at com.pwn9.filter.engine.FilterService.getAuthor(FilterService.java:252) ~[?:?]
	at com.pwn9.filter.bukkit.listener.PwnFilterPlayerListener.onPlayerChat(PwnFilterPlayerListener.java:62) ~[?:?]
	at com.pwn9.filter.bukkit.listener.PwnFilterPlayerListener.lambda$activate$32(PwnFilterPlayerListener.java:130) ~[?:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:517) [patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:499) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PlayerConnection.chat(PlayerConnection.java:1303) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PlayerConnection.a(PlayerConnection.java:1218) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PacketPlayInChat$1.run(PacketPlayInChat.java:39) [patched_1.9.2.jar:git-Paper-691]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_92]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_92]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_92]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_92]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_92]
Caused by: java.lang.InstantiationError
	at com.pwn9.filter.bukkit.BukkitAPI$1.load(BukkitAPI.java:67) ~[?:?]
	at com.pwn9.filter.bukkit.BukkitAPI$1.load(BukkitAPI.java:61) ~[?:?]
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3524) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2317) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2280) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2195) ~[patched_1.9.2.jar:git-Paper-691]
	... 20 more
[16:26:31] [Server thread/ERROR]: Could not pass event PlayerCommandPreprocessEvent to PwnFilter v3.9.1-SNAPSHOT-b101
com.google.common.util.concurrent.ExecutionError: java.lang.InstantiationError
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2199) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache.get(LocalCache.java:3934) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3938) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4821) ~[patched_1.9.2.jar:git-Paper-691]
	at com.pwn9.filter.bukkit.BukkitAPI.getAuthorById(BukkitAPI.java:75) ~[?:?]
	at com.pwn9.filter.bukkit.BukkitAPI.getAuthorById(BukkitAPI.java:46) ~[?:?]
	at com.pwn9.filter.engine.FilterService.getAuthor(FilterService.java:252) ~[?:?]
	at com.pwn9.filter.bukkit.listener.PwnFilterCommandListener.eventProcessor(PwnFilterCommandListener.java:96) ~[?:?]
	at com.pwn9.filter.bukkit.listener.PwnFilterCommandListener.lambda$activate$39(PwnFilterCommandListener.java:65) ~[?:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:517) [patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:502) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PlayerConnection.handleCommand(PlayerConnection.java:1380) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PlayerConnection.a(PlayerConnection.java:1193) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PacketPlayInChat.a(PacketPlayInChat.java:45) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PacketPlayInChat.a(PacketPlayInChat.java:5) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PlayerConnectionUtils$1.run(SourceFile:13) [patched_1.9.2.jar:git-Paper-691]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_92]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_92]
	at net.minecraft.server.v1_9_R1.SystemUtils.a(SourceFile:45) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.MinecraftServer.D(MinecraftServer.java:811) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.DedicatedServer.D(DedicatedServer.java:404) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.MinecraftServer.C(MinecraftServer.java:721) [patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.MinecraftServer.run(MinecraftServer.java:620) [patched_1.9.2.jar:git-Paper-691]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_92]
Caused by: java.lang.InstantiationError
	at com.pwn9.filter.bukkit.BukkitAPI$1.load(BukkitAPI.java:67) ~[?:?]
	at com.pwn9.filter.bukkit.BukkitAPI$1.load(BukkitAPI.java:61) ~[?:?]
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3524) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2317) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2280) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2195) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache.get(LocalCache.java:3934) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3938) ~[patched_1.9.2.jar:git-Paper-691]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4821) ~[patched_1.9.2.jar:git-Paper-691]
	at com.pwn9.filter.bukkit.BukkitAPI.getAuthorById(BukkitAPI.java:75) ~[?:?]
	at com.pwn9.filter.bukkit.BukkitAPI.getAuthorById(BukkitAPI.java:46) ~[?:?]
	at com.pwn9.filter.engine.FilterService.getAuthor(FilterService.java:252) ~[?:?]
	at com.pwn9.filter.bukkit.listener.PwnFilterPlayerListener.onPlayerChat(PwnFilterPlayerListener.java:62) ~[?:?]
	at com.pwn9.filter.bukkit.listener.PwnFilterPlayerListener.lambda$activate$32(PwnFilterPlayerListener.java:130) ~[?:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:517) ~[patched_1.9.2.jar:git-Paper-691]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:499) ~[patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PlayerConnection.chat(PlayerConnection.java:1303) ~[patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PlayerConnection.a(PlayerConnection.java:1218) ~[patched_1.9.2.jar:git-Paper-691]
	at net.minecraft.server.v1_9_R1.PacketPlayInChat$1.run(PacketPlayInChat.java:39) ~[patched_1.9.2.jar:git-Paper-691]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_92]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_92]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_92]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_92]
	... 1 more
```

**Comment by ptoal** [comment]
Wow... That's a new one for me.  Let me dig into this a bit.

**Comment by ptoal** [comment]
Ok.  I see what's wrong here.  There are two issues.  I will write a test, and fix them both soon.

**Comment by ptoal** [comment]
This is fixed in [Build #104](https://ci.sagely.ca/job/PwnFilter/104/)

**Comment by mibby** [comment]
Thanks. :)
